### PR TITLE
Add warning about using QueryBuilder with DI

### PR DESCRIPTION
**Pull request by sfroemkenjw**

### CODE REVIEW

File: Documentation/ApiOverview/Database/Connection/Index.rst [hunk 1]
```diff
@@ -81,6 +81,11 @@ Another way is to inject the :php:`Connection` object directly via
         :language: php
         :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
 
+..  warning::
+    Please do not include the `QueryBuilder` using dependency injection, as it
+    will contain various information about the last executed query (stateful).
+    These information can lead to side-effects if further queries will be
```
Comment (garvinhicking) [inline Documentation/ApiOverview/Database/Connection/Index.rst:87]:
```suggestion
    This information can lead to side-effects if further queries will be
```

File: Documentation/ApiOverview/Database/Connection/Index.rst [hunk 2]
```diff
@@ -81,6 +81,11 @@ Another way is to inject the :php:`Connection` object directly via
...
+    executed with the `QueryBuilder`.
```
Comment (garvinhicking) [inline Documentation/ApiOverview/Database/Connection/Index.rst:88]: Always create a new instance of this class via `GeneralUtility::makeInstance`.

**Review by garvinhicking — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Database/Connection/Index.rst (hunk 2 above)
Comment (garvinhicking) [inline Documentation/ApiOverview/Database/Connection/Index.rst:88]:
```suggestion
    executed with the `QueryBuilder`. Always create a new instance of this class via `GeneralUtility::makeInstance`.
```

Comment (linawolf) [reply Documentation/ApiOverview/Database/Connection/Index.rst:88]: I believe this is wrong as you should use the function of the connection pool: $queryBuilder = $this->connectionPool
            ->getQueryBuilderForTable('aTable');
            
            I would suggest we also link to that chapter: :ref:`Instantiation <t3coreapi:database-query-builder-instantiation>`

Comment (fe-hicking) [reply Documentation/ApiOverview/Database/Connection/Index.rst:88]: Of course, yeah, I forgot a few layers. My mindset was on "use makeInstance to retrieve the connectionpool", but that one is re-usable. Sorry.

**Review by brotkrueml — COMMENTED** [review]
Why do we need to mention query builder in a chapter about the Connection object?

**Comment by fe-hicking** [comment]
IMO connectionpool and querybuilder are tightly related, especially for common developers. Many people even confuse them, myself included 😂

**Comment by brotkrueml** [comment]
This should then be mentioned in the createQueryBuilder section IMHO:
https://docs.typo3.org/m/typo3/reference-coreapi/12.4/en-us/ApiOverview/Database/Connection/Index.html#database-connection-create-query-builder

**Comment by froemken** [comment]
I close this now:
* The problem does not occur for Connection
* The problem is only valid for QueryBuilder and there we already have a warning applied
* It does not make sense to add documentation to parts where it is not valid

Have a cool and sunny weekend

Froemmi

**Pull request closed without merging**